src/main/java/com/greghaskins/spectrum/internal/parameterized/Example.java

```diff
@@ -0,0 +1,35 @@
+package com.greghaskins.spectrum.internal.parameterized;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+public class Example<T> {
+
+  final String description;
+  private Consumer<T> blockRunner;
+
+  public Example(Consumer<T> blockRunner, Object... arguments) {
+    this.blockRunner = blockRunner;
+    this.description = describe(arguments);
+  }
+
+  public void runDeclaration(T block) {
+    this.blockRunner.accept(block);
+  }
+
+  @Override
+  public String toString() {
+    return description;
+  }
+
+  private static String describe(Object[] objects) {
+    return Arrays.stream(objects)
+        .map(o -> Optional.ofNullable(o)
+            .map(Object::toString)
+            .orElse("null"))
+        .collect(Collectors.joining("|", "|", "|"));
```

Comment:
Should there be spaces around the joining vertical bars? I don't know if that's in line with Cucumber proper, but [your screenshot](https://github.com/greghaskins/spectrum/pull/58#issuecomment-266140076) is hard to read for my eyes. `| 1 | 2 | 3 |` vs `|1|2|3|`.